docs/porting/connectivity/Layer3IP.md

```diff
@@ -0,0 +1,144 @@
+
+### The IP Layer3  memory manager
+
+The IP Layer3 memory manager class provides abstracted memory interface toward memory modules used in different network stacks. For the send and receive paths, data is transferred in memory buffers controlled through an `NetStackMemoryManager` object. The network stack using an L3IP driver provides it with a reference to the memory manager in use before powering up - this is constant as long as the L3IP is powered.
+
+Memory buffer chains store the data on the memory interface. The data passed in either direction either may be contiguous (a single-buffer chain) or may consist of multiple buffers. You can chain the buffers using a singly linked list.
+
+On the output call, the L3IP driver is given ownership of a buffer chain - it must free the chain when it has finished with the data. The data may or may not be contiguous. A driver can express alignment preferences for outgoing data, but the network stack is not required to meet these preferences, so a driver relying on alignment may need a slow path that copies data into an aligned (or contiguous) buffer.
+
+For reception, the L3IP driver must allocate memory from the `NetStackMemoryManager` to store the received packets - this is then passed to the link input callback, which frees it. By preference, you should allocate this memory using the pool, but if contiguous memory is necessary, you can allocate it from the heap.
+
+### IP Layer3  Interface
+
+All Mbed OS connectivity devices should provide an Mbed OS `NetworkInterface` implementation.
+
+To create and use an IP Layer3 interface `L3IPInterface` and  `L3IP` classes are needed. Moreover  L3IP derived driver class for specific target device is required, otherwise  linker error `undefined reference to L3IP::get_default_instance()` will stop build process.
+
+User app code can create IP Layer3 in way shown below
+
+   	NetworkInterface *l3interface;
+   	l3interface =new L3IPInterface(L3IP::get_default_instance(), OnboardNetworkStack::get_default_instance());
+   	l3interface->connect();
+
+This attaches the default network stack (usually LWIP - the other alternative is Nanostack) to the specified L3IP driver, and provides all the `NetworkInterface` and `NetworkStack` APIs.
+
+Example of target device driver class that needs to be implemented.
+
+
+	class Cellular_driver_L3IP : public L3IP {... }
+
+	Cellular_driver_L3IP &Cellular_driver_L3IP::get_instance()
+	{
+    	static Cellular_driver_L3IP l3ip_test_driver;
+    	return l3ip_test_driver;
+	}
+
+	// Weak so a module can override
+	MBED_WEAK L3IP &L3IP::get_default_instance()
+	{
+    	return Cellular_driver_L3IP::get_instance();
+	}
+
+Following driver class example can be found in mbed-os/TESTS/network/l3ip directory.
+
+
+### Being the default Interface
+Currently EMAC based  interfaces are created by automated factories and can be default for applications.
```

Comment:
This is not specific to EMAC.
Any interface type can be supplied as a default interface.

See https://os.mbed.com/docs/mbed-os/v5.11/apis/network-interfaces.html and scroll down for "Default network interface"

So re-phare this to something like: "In Mbed OS targets may provide default network interfaces through automated factory method"...

Reply:
Done